x-pack/legacy/plugins/apm/public/components/shared/charts/Timeline/Marker/ErrorMarker.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiPopover, EuiText } from '@elastic/eui';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import {
+  TRACE_ID,
+  TRANSACTION_ID
+} from '../../../../../../common/elasticsearch_fieldnames';
+import { useUrlParams } from '../../../../../hooks/useUrlParams';
+import { px, unit, units } from '../../../../../style/variables';
+import { asDuration } from '../../../../../utils/formatters';
+import { IWaterfallError } from '../../../../app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers';
+import { ErrorDetailLink } from '../../../Links/apm/ErrorDetailLink';
+import { Legend, Shape } from '../../Legend';
+
+interface Props {
+  mark: IWaterfallError;
+}
+
+const Popover = styled.div`
+  max-width: ${px(280)};
+`;
+
+const TimeLegend = styled(Legend)`
+  margin-bottom: ${px(unit)};
+`;
+
+const ErrorLink = styled(ErrorDetailLink)`
+  display: block;
+  margin: ${px(units.half)} 0 ${px(units.half)} 0;
+`;
+
+const Button = styled(Legend)`
+  height: 20px;
+  display: flex;
+  align-items: flex-end;
+`;
+
+export const ErrorMarker: React.FC<Props> = ({ mark }) => {
+  const { urlParams } = useUrlParams();
+  const [isPopoverOpen, showPopover] = useState(false);
+
+  const togglePopover = () => showPopover(!isPopoverOpen);
+
+  const button = (
+    <Button
+      clickable
+      color={theme.euiColorDanger}
+      shape={Shape.square}
+      onClick={togglePopover}
+    />
+  );
+
+  const error = mark.custom;
```

Comment:
Contrary to the waterfall item, I think it makes sense for the error document to be on the `custom` property of the `mark` interface 👍

Reply:
... you _might_ want to keep it as nested property - but that's not a big deal: 
```suggestion
  const { error } = mark.custom;
```